webapp/src/ts/modules/messages/messages.component.ts

```diff
@@ -0,0 +1,126 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { find as _find, isEqual as _isEqual } from 'lodash-es';
+import { combineLatest, Subscription } from 'rxjs';
+import { select, Store } from '@ngrx/store';
+
+import { MessageContactService } from '@mm-services/message-contact.service';
+import { GlobalActions } from '@mm-actions/global';
+import { MessagesActions } from '@mm-actions/messages';
+import { Selectors } from '@mm-selectors/index';
+import { ChangesService } from '@mm-services/changes.service';
+
+@Component({
+  templateUrl: './messages.component.html'
+})
+export class MessagesComponent implements OnInit, OnDestroy {
+  private globalActions: GlobalActions;
+  private messagesActions: MessagesActions;
+  subscriptions: Subscription = new Subscription();
+
+  loading = false;
+  loadingContent;
+  conversations = [];
+  selectedConversation;
+  error;
+
+  constructor(
+    private store: Store,
+    private changesService: ChangesService,
+    private messageContactService: MessageContactService
+  ) {
+    const subscription = combineLatest(
+      this.store.pipe(select(Selectors.getConversations)),
+      this.store.pipe(select(Selectors.getSelectedConversation)),
+      this.store.pipe(select(Selectors.getLoadingContent)),
+      this.store.pipe(select(Selectors.getMessagesError)),
+    ).subscribe(([
+      conversations,
+      selectedConversation,
+      loadingContent,
+      error,
+    ]) => {
+      this.selectedConversation = selectedConversation;
+      this.conversations = (conversations || []).sort((a, b) => b.date - a.date);
```

Comment:
I would prefer it if we do this in the reducer. It will also probably solve the error that this causes.

Reply:
Fixed